lib/learn_to_rank/ranker.rb

```diff
@@ -21,6 +22,43 @@ def ranks
     attr_reader :feature_sets
 
     def fetch_new_scores(examples)
+      endpoint = ENV["TENSORFLOW_SAGEMAKER_ENDPOINT"]
+      if endpoint
+        fetch_new_scores_from_sagemaker(examples, endpoint)
+      else
+        fetch_new_scores_from_serving(examples)
+      end
+    end
+
+    def fetch_new_scores_from_sagemaker(examples, endpoint)
+      begin
+        response = Aws::SageMakerRuntime::Client.new.invoke_endpoint(
+          endpoint_name: endpoint,
+          body: {
+            "signature_name": "regression",
+            "examples": examples,
+          }.to_json,
+          content_type: "application/json",
+          custom_attributes: "tfs-method=regress",
+        )
+      rescue Aws::SageMakerRuntime::Errors::ServiceError => e
+        logger.debug "SageMaker: #{e.message}"
+        log_error e.class.to_s
+        return nil
+      end
+
+      if response.nil? || response.body.blank?
+        logger.debug "SageMaker: No response from ranker!"
+        log_error "ranker_error"
```

Comment:
is it worth making this increment a different counter than the original one?  I'm a little undecided